dags/hivemind_etl_helpers/ingestion_pipeline.py

```diff
@@ -0,0 +1,66 @@
+import os
+
+from dotenv import load_dotenv
+from llama_index.core import MockEmbedding
+from llama_index.core.ingestion import (DocstoreStrategy, IngestionCache,
+                                        IngestionPipeline)
+from llama_index.core.node_parser import SemanticSplitterNodeParser
+from llama_index.storage.docstore.postgres import PostgresDocumentStore
+from llama_index.storage.kvstore.redis import RedisKVStore as RedisCache
+from llama_index.vector_stores.postgres import PGVectorStore
+from tc_hivemind_backend.db.credentials import load_postgres_credentials
+from tc_hivemind_backend.embeddings.cohere import CohereEmbedding
+
+from dags.hivemind_etl_helpers.src.db.gdrive.db_utils import setup_db
+
+
+class CustomIngestionPipeline:
+    def __init__(self, community_id: str, table_name: str,
+                 testing: bool = False):
+        load_dotenv()
+        postgres_credentials = load_postgres_credentials()
+        self.redis_host = os.getenv("REDIS_HOST")
```

Comment:
Please consider using a function to read the redis credentials. Another bounty hunter has implemented it, you can have a look

https://github.com/TogetherCrew/airflow-dags/blob/gdrive-ingestion/dags/hivemind_etl_helpers/src/utils/credentials.py#L32C1-L33C1

Also, please consider adding the related test cases of it

Here's the example
https://github.com/TogetherCrew/airflow-dags/blob/8dc5f4f17553431f69d74c67f7142b6097587620/dags/hivemind_etl_helpers/tests/integration/test_load_envs.py